airbyte-integrations/connectors/destination-s3-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/s3_v2/S3V2WriteTest.kt

```diff
@@ -47,21 +55,126 @@ abstract class S3V2WriteTest(path: String) :
     override fun testAppendSchemaEvolution() {
         super.testAppendSchemaEvolution()
     }
+
+    @Test
+    override fun testContainerTypes() {
+        super.testContainerTypes()
+    }
+
+    @Test
+    override fun testUnions() {
+        super.testUnions()
+    }
 }
 
-class S3V2WriteTestJsonUncompressed : S3V2WriteTest(S3V2TestUtils.JSON_UNCOMPRESSED_CONFIG_PATH)
+class S3V2WriteTestJsonUncompressed :
```

Comment:
tl;dr:
* json+csv don't stringify schemaless objects, don't promote unions to objects, and do preserve undeclared fields
* avro+parquet don't stringify, and don't preserve undeclared fields
    * avro doesn't promote unions
    * parquet does promote unions